registrations/tests/test_signup_group_put.py

```diff
@@ -182,6 +183,112 @@ def test_registration_admin_can_update_signup_group(
     assert signup1.last_modified_by_id is None
 
 
+@pytest.mark.django_db
+def test_contact_person_can_update_signup_group_when_strongly_identified(
+    api_client, registration
+):
+    user = UserFactory()
+    api_client.force_authenticate(user)
+
+    signup_group = SignUpGroupFactory(registration=registration)
+    signup = SignUpFactory(signup_group=signup_group, registration=registration)
+    contact_person = SignUpContactPersonFactory(signup_group=signup_group, user=user)
+
+    new_signup_name = "Edited name"
+
+    signup_group_data = {
+        "registration": registration.id,
+        "extra_info": new_signup_group_extra_info,
+        "signups": [
+            {
+                "id": signup.id,
+                "first_name": new_signup_name,
+            },
+        ],
+    }
+
+    assert signup_group.extra_info is None
+    assert signup_group.last_modified_by_id is None
+
+    assert signup.first_name != new_signup_name
+    assert signup.last_modified_by_id is None
+
+    with patch(
+        "helevents.models.UserModelPermissionMixin.token_amr_claim",
+        new_callable=PropertyMock,
+        return_value=["suomi_fi"],
+    ) as mocked:
+        assert_update_signup_group(
+            api_client,
+            signup_group.id,
+            signup_group_data,
+        )
+        assert mocked.called is True
+
+    signup_group.refresh_from_db()
+    del signup_group.extra_info
```

Comment:
should this be here? and if should then why?

Reply:
This is to refresh the cached property and ensure that the correct value is being checked

Reply:
Oh right! Should we still invalidate the property in the logic level if that is something of a feature we want to have. Instead of invalidating it here in the test just to get the test working 🤔 
(btw I ran the test locally and it passed without the cache but  I guess since cache it would flicker in some cases. )

Reply:
It could be useful as a general feature to add the invalidation to the `save` method of the objects that use these cached properties or maybe add it to `refresh_from_db`. In the API views we generally don't mind if we have a cached value as long as we are aware of this in the rest of the logic, and `get_object`will fetch a fresh instance without the cached value(s) AFAIK. So perhaps for this case it's OK to invalidate it in the test as it is only to check that the DB value has changed correctly?

Reply:
Sounds reasonable 🤝